syntax = "proto3";

package greet;

// The greeting service definition.

service Greeter {
	
    rpc VoidTeste (VoidRequest) returns (VoidReply);
	
	  rpc LongSimplesTeste (LongRequest) returns (LongReply);
	
    rpc LongComplexoTeste (LongComplexRequest) returns (LongComplexReply);

    rpc StringTeste (StringRequest) returns (StringReply);

    rpc ClassTeste (ClasseRequest) returns (ClasseReply);
    
}

//Void
message VoidRequest {
    
}
message VoidReply {
  
}

// Long
message LongRequest {
  int64 value = 1;
}

message LongReply {
  int64 result = 1;
}

// Long 8 Argumentos 1 resposta
message LongComplexRequest {
  int64 arg1 = 1;
  int64 arg2 = 2;
  int64 arg3 = 3;
  int64 arg4 = 4;
  int64 arg5 = 5;
  int64 arg6 = 6;
  int64 arg7 = 7;
  int64 arg8 = 8;
}

// Definição da mensagem de resposta do servidor
message LongComplexReply {
  int64 result = 1;
}

//String
message StringRequest {
  string name = 1;
}

message StringReply {
  string message = 1;
}

// Classe
message Pessoa {
  string nome = 1;
  int32 idade = 2;
  float altura = 3;
}

message ClasseRequest {
  int32 id = 1;
}

message ClasseReply {
  Pessoa person = 1;
}
